### PR TITLE
[DCWL-2007] Play 3.0 Upgrade

### PR DESCRIPTION
**Pull request by anna-shen-hmrc**
Please ensure we are following our Ways of Working for Pull Requests: https://confluence.tools.tax.service.gov.uk/display/BTL/Ways+of+Working+for+Pull+Requests

### CODE REVIEW

File: app/uk/gov/hmrc/customs/notification/controllers/dynamicservice/DynamicServiceConfigurationController.scala [hunk 1]
```diff
@@ -17,7 +17,7 @@
 package uk.gov.hmrc.customs.notification.controllers.dynamicservice
 
 import com.google.inject.Inject
-import play.api.libs.json.{JsError, JsSuccess, Json}
+import play.api.libs.json._
```
Comment (gavinharris43) [inline app/uk/gov/hmrc/customs/notification/controllers/dynamicservice/DynamicServiceConfigurationController.scala:20]:
```suggestion
import play.api.libs.json.*
```

File: app/uk/gov/hmrc/customs/notification/domain/ClientNotification.scala [hunk 2]
```diff
@@ -17,18 +17,19 @@
 package uk.gov.hmrc.customs.notification.domain
 
 import org.bson.types.ObjectId
-import org.joda.time.DateTime
-import play.api.libs.json.{Format, Json}
-import uk.gov.hmrc.mongo.play.json.formats.{MongoFormats, MongoJodaFormats}
+import play.api.libs.json.{Format, Json, OFormat}
+import uk.gov.hmrc.mongo.play.json.formats.{MongoFormats, MongoJavatimeFormats}
+
+import java.time.Instant
```
Comment (gavinharris43) [inline app/uk/gov/hmrc/customs/notification/domain/ClientNotification.scala:23]:
```suggestion
import play.api.libs.json.{Format, Json, OFormat}
import uk.gov.hmrc.mongo.play.json.formats.{MongoFormats, MongoJavatimeFormats}
import java.time.Instant
```

File: app/uk/gov/hmrc/customs/notification/services/CustomsNotificationService.scala [hunk 3]
```diff
@@ -51,7 +50,7 @@ class CustomsNotificationService @Inject()(logger: NotificationLogger,
     implicit val hasId: RequestMetaData = metaData
     val notificationWorkItem = NotificationWorkItem(metaData.clientSubscriptionId,
       ClientId(apiSubscriptionFields.clientId),
-      Some(metaData.startTime.toDateTime),
+      Some(metaData.startTime.toInstant),
```
Comment (gavinharris43) [inline app/uk/gov/hmrc/customs/notification/services/CustomsNotificationService.scala:53]:
```suggestion
      Some(metaData.startTime.toInstant),
      //TODO replace this ..... test
```

File: app/uk/gov/hmrc/customs/notification/services/CustomsNotificationService.scala [hunk 4]
```diff
@@ -51,7 +50,7 @@ class CustomsNotificationService @Inject()(logger: NotificationLogger,
...
       Notification(Some(metaData.notificationId), metaData.conversationId, buildHeaders(metaData), xml.toString, MimeTypes.XML))
```
Comment (gavinharris43) [inline app/uk/gov/hmrc/customs/notification/services/CustomsNotificationService.scala:54]:
```suggestion
      Some(metaData.startTime.toInstant),
      Notification(Some(metoData.notificationId), metaData.conversationId, buildHeaders(metaData), xml.toString, MimeTypes.XML))
```

File: app/uk/gov/hmrc/customs/notification/controllers/dynamicservice/DynamicServiceConfigurationController.scala (hunk 1 above)
Comment (anna-shen-hmrc) [reply app/uk/gov/hmrc/customs/notification/controllers/dynamicservice/DynamicServiceConfigurationController.scala:20]: Tried but it has compile error. Anything that I missed?
image

**Review by IanWood-HMRC — APPROVED** [review]

**Pull request merged**